src/main/java/com/cs3365/rpgdapi/weapons/WeaponsController.java

```diff
@@ -47,4 +48,15 @@ public WeaponsController(WeaponsService weaponsService) {
         return ResponseEntity.created(location)
                              .body("");
     }
+    
+    @RequestMapping(value = "/files/{id}")
+    public ResponseEntity<WeaponEntity> get(@PathVariable("id") UUID identifier)
+    {
+        try {
+            return ResponseEntity.ok(weaponsService.findWeapon(identifier));
+        }
+        catch(WeaponsException e) {
+            return ResponseEntity.badRequest().header("Could not retrieve the weapon from the database").body(null);
```

Comment:
You wouldn't want to use the `header()` method here. You want to put that message in the `body()` method.